### ISSUE TITLE
MinGW error: undefined reference to `YAML::detail::node::m_amount'

### ISSUE THREAD

**Issue by ljx0305** [jbeder/yaml-cpp#1026]
CMake version: 3.21.1
MinGW g++ version: 4.8.1
yaml-cpp version:  yaml-cpp-master 

E:\yaml-cpp-master\build>C:\Qt\4.8.7\mingw482_32\bin\mingw32-make
Consolidate compiler generated dependencies of target yaml-cpp
[ 18%] Built target yaml-cpp
Consolidate compiler generated dependencies of target gtest
[ 19%] Built target gtest
Consolidate compiler generated dependencies of target gmock
[ 20%] Built target gmock
Consolidate compiler generated dependencies of target yaml-cpp-tests
[ 21%] Linking CXX executable yaml-cpp-tests.exe
CMakeFiles\yaml-cpp-tests.dir/objects.a(error_messages_test.cpp.obj):error_messages_test.cpp:(.text$_ZN4YAML6detail4node9push_backERS1_St10shared_ptrINS0_13memory_holderEE[__ZN4YAML6detail4node9push_backERS1_St10shared_ptrINS0_13memory_holderEE]+0x69): undefined reference to `YAML::detail::node::m_amount'
collect2.exe: error: ld returned 1 exit status
test\CMakeFiles\yaml-cpp-tests.dir\build.make:305: recipe for target 'test/yaml-cpp-tests.exe' failed
mingw32-make[2]: *** [test/yaml-cpp-tests.exe] Error 1
CMakeFiles\Makefile2:1071: recipe for target 'test/CMakeFiles/yaml-cpp-tests.dir/all' failed
mingw32-make[1]: *** [test/CMakeFiles/yaml-cpp-tests.dir/all] Error 2
Makefile:144: recipe for target 'all' failed
mingw32-make: *** [all] Error 2

**Comment by ScottBailey** [comment]
Same problem with:
- CMake v3.21.3
- MSVC 19.24.28314.0
- yaml-cpp tag: yaml-cpp-0.7.0

However, after investigation, this is fixed for me in or before `edadfecdc644e2a75ef6074e7afe74a3ebb44078`.

Files are `include/yaml-cpp/dll.h` and `include/yaml-cpp/node/detail/node.h`. When the latest are copied into a v0.7.0 repo, the tests will build and run.

I believe this issue should be closed.

**Comment by PatrceBrunelleCimel** [comment]
It is indeed fixed by commit 4564d9a1314e8dbd01ba769b94133b9eba4f36ad on file include/yaml-cpp/node/detail/node.h

It is just the symbol m_amount which is not exported.
```
-  static std::atomic<size_t> m_amount;
+  static YAML_CPP_API std::atomic<size_t> m_amount;
```

**Comment by HunterZ** [comment]
Any idea when a new release be available with this fix? It's 6 months later and a broken yaml-cpp package is still being shipped by MSYS MINGW64 because 0.7.0 is still the latest release version :(
